packages/flutter/lib/src/material/text_selection.dart

```diff
@@ -20,64 +23,390 @@ const double _kHandleSize = 22.0;
 const double _kToolbarScreenPadding = 8.0;
 const double _kToolbarHeight = 44.0;
 // Padding when positioning toolbar below selection.
-const double _kToolbarContentDistanceBelow = 16.0;
+const double _kToolbarContentDistanceBelow = _kHandleSize - 2.0;
 const double _kToolbarContentDistance = 8.0;
 
 /// Manages a copy/paste text selection toolbar.
-class _TextSelectionToolbar extends StatelessWidget {
+class _TextSelectionToolbar extends StatefulWidget {
   const _TextSelectionToolbar({
     Key key,
     this.handleCut,
     this.handleCopy,
     this.handlePaste,
     this.handleSelectAll,
+    this.isAbove,
   }) : super(key: key);
 
   final VoidCallback handleCut;
   final VoidCallback handleCopy;
   final VoidCallback handlePaste;
   final VoidCallback handleSelectAll;
 
+  // When true, the toolbar fits above its anchor and will be positioned there.
+  final bool isAbove;
+
+  // Returns true iff the menu items that this widget renders will produce a
+  // different width than that of oldWidget. Width depends on the existence of
+  // callbacks for their respective buttons.
+  bool menuWidthChanged(_TextSelectionToolbar oldWidget) {
+    return (handleCut == null) != (oldWidget.handleCut == null)
+      || (handleCopy == null) != (oldWidget.handleCopy == null)
+      || (handlePaste == null) != (oldWidget.handlePaste == null)
+      || (handleSelectAll == null) != (oldWidget.handleSelectAll == null);
+  }
+
+  @override
+  _TextSelectionToolbarState createState() => _TextSelectionToolbarState();
+}
+
+class _TextSelectionToolbarState extends State<_TextSelectionToolbar> {
+  final GlobalKey _containerKey = GlobalKey();
+  final GlobalKey _moreButtonKey = GlobalKey();
+
+  // The width of all of the items that are being rendered in the closed
+  // toolbar, including the more button if visible.
```

Comment:
the "more" button, if visible

Reply:
No longer relevant with the RenderBox approach.